packages/core/src/blockchain/storage-layer.ts

```diff
@@ -111,12 +111,25 @@ export class RemoteStorageLayer implements StorageLayerProvider {
       }
 
       if (this.#db) {
-        // batch fetch storage values and save to db, they may be used later
-        this.#api.getStorageBatch(prefix as HexString, batch as HexString[], this.#at as HexString).then((storage) => {
-          for (const [key, value] of storage) {
-            this.#db?.saveStorage(this.#at as HexString, key as HexString, value)
+        // filter out keys that are not in the db]
+        const newBatch: HexString[] = []
+
+        for (const key of batch) {
+          const res = await this.#db.queryStorage(this.#at, key)
```

Comment:
it will be better if we add additional api to check existing keys in batches

Reply:
it is not too slow to make 1000 requests to sqlite but yeah we can improve it #899